adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -102,91 +100,73 @@ - (void)presentFromViewController:(nonnull UIViewController *)viewController {
     GADMAdapterUnityMapTableRemoveObjectForKey(_placementInUseRewarded, self->_placementID);
   });
 
-  if (![UnityAds isReady:_placementID]) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorShowAdNotReady, @"Failed to show Unity Ads rewarded video.");
-    [_adEventDelegate didFailToPresentWithError:error];
-    return;
-  }
   [_adEventDelegate willPresentFullScreenView];
-  [UnityAds addDelegate:self];
-  [UnityAds show:viewController placementId:_placementID];
+  [UnityAds show:viewController placementId:_placementID showDelegate:self];
 }
 
-#pragma mark - UnityAdsExtendedDelegate Methods
+- (void)cleanup {
+  // Clean up references to prevent a memory leak where the object can not be cleand up
+  // due to holding on to the references.
+  _adEventDelegate = nil;
+  _adLoadCompletionHandler = nil;
```

Comment:
I prefer not to set completion handlers to nil due to possible crash if this is ever called without a nil check. If https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/CONTRIBUTING.md#best-practices #7 is implemented on this class, the original completion handler is nilled out once we invoke the handler the first time, and there should be no need to set completion handler to nil.

Reply:
Thanks for the link @ericleich.  The suggested change fix the leak.  I will push an update shortly.